tithemi/assignment1/Q1.cpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <algorithm>
+
+std::unordered_map<char, int> count_characters(const std::string &s, bool is_case_sensitive = false) {
+    std::unordered_map<char, int> characters;
+
+    if (is_case_sensitive) {
+        for (const char c : s) {
+            characters[c] += 1;
+        }
+    } else {
+        for (const char c : s) {
+            characters[std::tolower(c)] += 1;
+        }
+    }
```

Comment:
[optional]: Generally, the code is more readable if you branch within for loops and not the other way around, example:

```
for (const char c : s) {
    if (is_case_sensitive) { characters[c] += 1; }
    else { characters[std::tolower(c)] += 1; }
}
```

It's also shorter :)